mock-bank-system/src/main/java/com/centiglobe/mockbanksystem/presentation/BankController.java

```diff
@@ -75,34 +42,48 @@
     @PostMapping("/handle")
     public ResponseEntity<String> handleMessage(@RequestBody String encodedMx) {
         LOGGER.info("Mock bank system handling ISO 20022 message.");
-        String from;
-        String to;
+
         try {
+            MxPacs00200111 responseMessage;
+
             AbstractMX mx = AbstractMX.parse(encodedMx);
             if (mx == null)
                 return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Bad message.");
 
-            BusinessAppHdrV02 header = ((BusinessAppHdrV02) mx.getAppHdr());
-            if ("pacs".equals(mx.getBusinessProcess())) {
-                MxPacs00800109 pacs = (MxPacs00800109) mx;
-                for (CreditTransferTransaction43 transaction : pacs.getFIToFICstmrCdtTrf().getCdtTrfTxInf()) {
-                    // Ignores the currency and any other metadata
-                    balance += transaction.getIntrBkSttlmAmt().getValue().doubleValue();
-                    LOGGER.info("Received payment, new balance is " + balance + ".");
-                }
+            if ("pacs".equals(mx.getBusinessProcess())){
+                MxPacs00800109 pacs008 = (MxPacs00800109) mx;
+
+                responseMessage = createResponseForPacs008(pacs008);

```

Comment:
Maybe there should be a check that mx really can be cast in case it is for example a pacs.002 message?

Reply:
I will modify it to check for the exact version of the pacs message 👍